packages/core/react/src/WalletProvider.tsx

```diff
@@ -239,6 +239,16 @@ export const WalletProvider: FC<WalletProviderProps> = ({
         [adapter, handleError, connected]
     );
 
+    // Send a versioned transaction using the provided connection
+    const sendVersionedTransaction = useCallback(
```

Comment:
I think we can just roll all these into the existing API and keep most changes at the adapter level. If we conditionally type the methods' `transaction` inputs, in the context's `sendTransaction`, `signTransaction`, and `signAllTransactions` methods, we'll have to perform the same conditional typing against `adapter.supportedTransactionVersions`.